services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/consumer/AdminConsumptionTask.java

```diff
@@ -925,6 +927,25 @@ private Pair<Long, AdminOperation> extractExecutionIdAndAndAdminOperation(Defaul
     return new Pair<>(executionId, adminOperation);
   }
 
+  private AdminOperation deserializeAdminOperation(Put put) {
+    boolean isMessageWithFutureProtocolVersion = put.schemaId > LATEST_SCHEMA_ID_FOR_ADMIN_OPERATION;
+    if (admin.isAdminOperationSystemStoreEnabled() && isMessageWithFutureProtocolVersion) {
+      deserializer.fetchAndStoreSchemaIfAbsent(admin, put.schemaId);
+      AdminOperation adminOperation = deserializer.deserialize(put.putValue, put.schemaId);
+      stats.recordAdminMessagesWithFutureProtocolVersionCount();
+      String warningMessage = String.format(
+          "Deserialized admin operation with a newer protocol version. Schema id: %d, operation: %s",
+          put.schemaId,
+          adminOperation);
+
+      if (!REDUNDANT_LOGGING_FILTER.isRedundantException(warningMessage)) {
```

Comment:
The throttling won't work because each operation is different.

Reply:
Discussed offline with @huangminchn , this throttling will be useful when we have a stuck message that failed to process and this log line won't overload the log file.